common-build.gradle

```diff
@@ -47,3 +47,133 @@ task syncVersionAndroid  {
+
+        // actually clone the repo, if necessary
+        if (cloneRepo) {
+            exec {
+                commandLine "git", "clone", "--branch", "${airshipProperties.iosVersion}", "https://github.com/urbanairship/ios-library", "${iosTempRepo}"
+            }
+        }
+
+        // SourceKitten
+        println "Running Sourcekitten"
+
+        // get path to the module's source
+        def moduleSourcePath = ""
+        if ( iosSdkModuleName.endsWith("Extension") ) {
+            moduleSourcePath="AirshipExtensions/${iosSdkModuleName}/Source"
+        } else {
+            moduleSourcePath="Airship/${iosSdkModuleName}/Source"
+        }
+
+        // get path to the module's header
+        def moduleHeader = ""
+        if ( iosSdkModuleName == "AirshipCore" ) {
+            moduleHeader = "${moduleSourcePath}/Common/${iosSdkModuleName}.h"
+        } else {
+            moduleHeader ="${moduleSourcePath}/${iosSdkModuleName}.h"
+        }
+
+        // get path to the SDK
+        def out = new ByteArrayOutputStream()
+        exec {
+            commandLine "xcrun", "--show-sdk-path", "--sdk", "iphonesimulator"
+            standardOutput = out
+        }
+        def sdkPath = out.toString().replaceAll("\\s","")
+
+        // arguments for the sourcekitten command
+        def skArgs = [ "doc",
+                       "--objc",
+                       moduleHeader,
+                       "--",
+                       "-x",
+                       "objective-c",
+                       "-isysroot", sdkPath,
+                       "-I", "${iosTempRepo}",
+                       "-I", "${iosTempRepo}/Airship/AirshipCore/Source/common",
+                       "-I", "${iosTempRepo}/Airship/AirshipCore/Source/ios",
+                       "-I", "${iosTempRepo}/${moduleSourcePath}",
+                       "-fmodules",
+                       "-D", "UA_USE_MODULE_IMPORT=0" ]
+
+        // manually add all folders in the module's Source directory, as I could not figure out how to get "-I ${iosTempRepo}/Airship/${MODULE_NAME}/Source/**"" to work
+        out.reset()
+        exec {
+            workingDir iosTempRepo
+            commandLine "find", moduleSourcePath, "-type", "d", "-mindepth", "1"
+            standardOutput = out
+        }
+        def moduleFolders = out.toString().split('\n')
+        moduleFolders.each {
+            skArgs.addAll("-I", "${iosTempRepo}/${it}");
+        }
+
+        def skParse = "${iosTempRepo}/sourcekitten_parse.${iosSdkModuleName}.json"
+
+        // run SourceKitten
+        exec {
+            workingDir iosTempRepo
+            executable "sourcekitten"
+            args skArgs
+            standardOutput new FileOutputStream(skParse)
+        }
+
+        // Xamboni
+        println "Running Xamboni"
+
+        def xamboniTempRepo = new File("${tmpDir}/xamboni")
+        xamboniTempRepo.deleteOnExit()
+
+        boolean cloneXamboniRepo = false
+        if ( !xamboniTempRepo.exists() ) {
```

Comment:
what if the cloned xamobni repo is out of date?

Reply:
That could happen, if the gradle daemon has been running for a long time. Maybe I should pull, also, or get Xamboni in a different way.

Reply:
I added a `clean` task to run if there is any doubt as to the recency of the repo. `/gradlew clean generateIOSBindings` will remove the temporary iOS SDK and Xamboni repos, and then run `generateIOSBindings`, which will clone the repos anew.